src/tests/ftest/deployment/agent_failure_isolate.py

```diff
@@ -0,0 +1,184 @@
+        ior_cmd.test_file.update(testfile)
+
+        manager = get_job_manager(
+            test=self, class_name="Mpirun", job=ior_cmd, subprocess=self.subprocess,
+            mpi_type="mpich")
+        manager.assign_hosts(clients, self.workdir, self.hostfile_clients_slots)
+        ppn = self.params.get("ppn", '/run/ior/client_processes/*')
+        manager.ppn.update(ppn, 'mpirun.ppn')
+        manager.processes.update(None, 'mpirun.np')
+
+        try:
+            ior_output = manager.run()
+            results[job_num] = [True]
+            # For debugging.
+            results[job_num].extend(IorCommand.get_ior_metrics(ior_output))
+            # We'll verify the error message.
+            results[job_num].append(ior_output.stderr_text)
+        except CommandFailure as error:
+            results[job_num] = [False, "IOR failed: {}".format(error)]
+
+    def test_agent_failure_isolation(self):
+        """Jira ID: DAOS-9385.
+
+        1. Create a pool and a container.
+        2. Run IOR from the two client nodes.
+        3. Stop daos_agent process while IOR is running on one of the clients.
+        4. Wait until both of the IOR ends.
+        5. Check that there's no error on both of the clients. (No error occurs if there's
+        another agent running.)
+        6. On the killed client, verify journalctl shows the log that the agent is
+        stopped.
+        7. On the other client where agent is still running, verify that the journalctl
+        doesn't show that the agent is stopped.
+        8. Restart both daos_agent.
+        9. Run IOR again from both clients. It should succeed this time without any
+        error.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=deployment,fault_management,agent_failure
+        :avocado: tags=agent_failure_isolation
+        """
+        # 1. Create a pool and a container.
+        self.add_pool()
+        self.add_container(self.pool)
+
+        agent_hosts = self.agent_managers[0].hosts
+        self.log.info("agent_hosts = %s", agent_hosts)
+        agent_host_keep = agent_hosts[0]
+        agent_host_kill = agent_hosts[1]
+
+        # 2. Run IOR from the two client nodes.
+        ior_results = {}
+        job_num_keep = 1
+        job_num_kill = 2
+        self.log.info("Run IOR with thread")
+        thread_1 = threading.Thread(
+            target=self.run_ior_collect_error,
+            args=[ior_results, job_num_keep, "test_file_1", [agent_host_keep]])
+        thread_2 = threading.Thread(
+            target=self.run_ior_collect_error,
+            args=[ior_results, job_num_kill, "test_file_2", [agent_host_kill]])
+
+        self.log.info("Start IOR 1 (thread)")
+        thread_1.start()
+        thread_2.start()
+
+        # We need to stop daos_agent while IOR is running, so need to wait for a few
+        # seconds for IOR to start.
+        self.log.info("Sleep 3 sec")
+        time.sleep(3)
```

Comment:
Should we use 5 seconds, similar to `agent_failure.py`?

Reply:
Yes. Fixed.